genie-agent/src/main/java/com/netflix/genie/agent/cli/ExecCommand.java

```diff
@@ -69,20 +65,15 @@ public ExitCode run() {
             sun.misc.Signal.handle(s, signal -> handleTerminationSignal());
         }
 
-        log.info("Running job state machine");
-        stateMachine.start();
-
-        final States finalState;
+        log.info("Starting job execution");
         try {
-            finalState = stateMachine.waitForStop();
+            this.stateMachine.run();
         } catch (final Exception e) {
-            log.warn("Job state machine execution failed", e);
-            throw new RuntimeException("Job execution error", e);
+            log.error("Job state machine execution failed: {}", e.getMessage());
```

Comment:
are you purposely not adding the exception to this log message?

Reply:
Yes, this is happening pretty much in `main()` which prints the full trace anyway right after.